app/src/main/java/com/github/sdpteam15/polyevents/item_request/ItemRequestAdapter.kt

```diff
@@ -0,0 +1,122 @@
+     */
+    inner class ItemViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {
+
+        private val itemName = view.findViewById<TextView>(R.id.id_item_name)
+        private val itemQuantity = view.findViewById<EditText>(R.id.id_item_quantity)
+
+        /**
+         * Binds the value of the item to the layout of the item tab
+         */
+        @RequiresApi(Build.VERSION_CODES.O)
+        fun bind(item: Pair<String, Int>) {
+            itemName.text =
+                view.context.getString(R.string.item_name_quantity_text, item.first, item.second)
+
+            // Set initial quantity to 0
+            itemQuantity.setText("0")
+            itemQuantity.addTextChangedListener(object : TextWatcher {
+                override fun beforeTextChanged(
+                    s: CharSequence?,
+                    start: Int,
+                    count: Int,
+                    after: Int
+                ) {
+                    // Do nothing
+                }
+
+                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                    // Do nothing
+                }
+
+                override fun afterTextChanged(s: Editable?) {
+                    if (s.toString().isNotEmpty()) {
+                        val value = s.toString().toInt()
+                        when {
+                            value < 0 -> {
+                                // Value set it negative, change it to 0
+                                // and inform the user
+                                itemQuantity.setText("0")
+
+                                Toast.makeText(
+                                    view.context,
+                                    view.context.getString(R.string.item_quantity_positive_text),
+                                    Toast.LENGTH_LONG
+                                )
+                                    .show()
+                            }
+                            value > item.second -> {
+                                // The quantity set is too high, set it to the max quantity
+                                // available and inform the user
+                                val maxQuantity = item.second
+                                itemQuantity.setText(maxQuantity.toString())
+
+                                // Update the list with the max quantity available
+                                onItemQuantityChangeListener(item.first, maxQuantity)
+
+                                Toast.makeText(
+                                    view.context,
+                                    view.context.getString(
+                                        R.string.max_item_quantity_text,
+                                        maxQuantity.toString(),
+                                        itemName.text
+                                    ),
+                                    Toast.LENGTH_LONG
+                                )
```

Comment:
Can we extract a method as they are multiple times the same thing ?

Reply:
Good idea ! I will put it in the Helper Functions